src/System.IO.Pipelines.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -116,6 +116,23 @@ public IntPtr GetBufferId(IntPtr address, out long startAddress)
             return id;
         }
 
+        public unsafe RioBufferSegment GetSegmentFromMemory(Buffer<byte> memory)
+        {
+            var pin = memory.Pin();
```

Comment:
Add a comment about why it's ok to unpin here because the memory needs to be pinned already? How do we enforce that though? There's no way to know if the backing memory is already pinned...